### ISSUE TITLE
S4261 should expect Async suffix when method is returning IAsyncEnumerable

### ISSUE THREAD

**Issue by lg2de** [SonarSource/sonar-dotnet#3264]
### Description
The rule **S4261** should expect `Async` suffix when method is returning `IAsyncEnumerable<T>`.

### Repro steps

```cs
public interface IInterface
{
    IAsyncEnumerable<int> GetValuesAsync(IEnumerable<int> identifiers);
}
```

### Expected behavior
The rule should not raise. The method is an asynchronous method, isn't it?
So, the method should be named using `Async`.

### Actual behavior
`warning S4261: Remove the 'Async' suffix to the name of this method.`

### Known workarounds
N/A

### Related information
* C#/VB.NET Plugins Version: SonarAnalyzer.CSharp 8.5.0.15942
* Visual Studio Version: 16.5.1

**Comment by pavel-mikula-sonarsource** [comment]
Hi @lg2de ,

Your `GetValuesAsync` is synchronously creating object that could be later asynchronously iterated. So it is synchronous method and should not contain `Async` suffix. Rule behaves as expected here.

Looking at the signature, I would say `GetAsyncValues` would be more precise name.
